command/agent/agent.go

```diff
@@ -822,6 +963,27 @@ func (a *Agent) Shutdown() error {
 	if a.shutdown {
 		return nil
 	}
+	a.logger.Println("[INFO] agent: Requesting shutdown")
+
+	// Stop all API endpoints
+	a.logger.Println("[INFO] agent: Stopping DNS endpoints")
+	for _, srv := range a.dnsServers {
+		srv.Shutdown()
+	}
+	for _, srv := range a.httpServers {
+		a.logger.Println("[INFO] agent: Stopping HTTP endpoint", srv.Addr)
```

Comment:
No, since it now logs one line per endpoint with the specific address of the endpoint.